ee/packages/pdf-worker/src/strategies/ChatTranscript.ts

```diff
@@ -10,8 +12,41 @@ export class ChatTranscript implements IStrategy {
 		return !previous || !moment(current.ts).tz(timezone).isSame(previous.ts, 'day');
 	}
 
-	private parserMessages(messages: PDFMessage[], dateFormat: string, timeAndDateFormat: string, timezone: string): PDFMessage[] {
-		return messages.map((message, index, arr) => {
+	private async parseSystemMessage(messages: PDFMessage[]): Promise<PDFMessage[]> {
+		return Promise.all(
+			messages.map(async (message: PDFMessage) => {
+				if (!message.t) {
+					return message;
+				}
```

Comment:
```suggestion
			messages.filter((message) => !!message.t).map(async (message: PDFMessage) => {
```
Filter outside the async to reduce promises created